pkg/reconciler/certificate/resources/cert_manager_certificate.go

```diff
@@ -111,18 +110,38 @@ func MakeCertManagerCertificate(cmConfig *config.CertManagerConfig, knCert *v1al
 
 	dnsNames = append(dnsNames, knCert.Spec.DNSNames...)
 
+	var certType string
+	var ok bool
+	// default to CertificateExternalDomain if label is not set
+	if certType, ok = knCert.Labels[networking.CertificateTypeLabelKey]; !ok {
+		certType = string(netapi.CertificateExternalDomain)
+	}
```

Comment:
```suggestion
        // default to CertificateExternalDomain
        certType := netapi.CertificateExternalDomain
        if val, ok := knCert.Labels[networking.CertificateTypeLabelKey]; ok {
                certType = netapi.CertificateType(val)
        }
```

This is just my preference but we can cast the string label into `CertificateType`. Then, we don't need cast it to string every single type below.

```go
	switch certType {
	case netapi.CertificateClusterLocalDomain:
        ...
	case netapi.CertificateSystemInternal:
        ...
```